pysolarfocus/__init__.py

```diff
@@ -429,189 +323,140 @@ def update(self):
-        )
-        return self.__write_registers(HP_REGMAP_HOLDING["OUTDOOR_TEMPERATURE_EXTERNAL"]["addr"],temp_scaled)
+        return self.__write_register(self.heatpump.outdoor_temperature_external,temperature)
     
     def pv_set_smart_meter(self, value: int) -> bool:
         """Set Smart Meter"""
-        return self.__write_registers(PV_REGMAP_HOLDING["SMART_METER"]["addr"],value)
+        return self.__write_register(self.photovoltaic.smart_meter,value)
 
     def pv_set_photovoltaic(self, value: int) -> bool:
         """Set Photovoltaic"""
-        return self.__write_registers(PV_REGMAP_HOLDING["PHOTOVOLTAIC"]["addr"], value)
+        return self.__write_register(self.photovoltaic.photovoltaic, value)
 
 
     def pv_set_grid_im_export(self, value: int) -> bool:
         """Set Photovoltaic"""
-        return self.__write_registers(PV_REGMAP_HOLDING["GRID_IM_EXPORT"]["addr"],value)
-
-
-    def __write_registers(self,address:int, value:int, check_connection:bool = True) -> bool:
-        """Internal methode to write registers to the modbus server"""
+        return self.__write_register(self.photovoltaic.grid_im_export,value)
+    
+    def __write_register(self,data_value:DataValue, value:int, check_connection:bool = True) -> bool:
+        """Internal methode to write a value to the modbus server"""
         if check_connection and not self.is_connected:
             logging.error("Connection to modbus is not established!")
             return False
         try:
-            response = self._conn.write_registers(address, [value], unit=self._slave_id)
+            scaled = int(data_value.reverse_scale(value))
+            response = self._conn.write_registers(data_value._absolut_address, [scaled], unit=self._slave_id)
             if response.isError():
-                logging.error(f"Error writing value={value} to register: {address}: {response}")
+                logging.error(f"Error writing value={value} to register: {data_value._absolut_address}: {response}")
                 return False
         except Exception:
-            logging.exception(f"Eception while writing value={value} to register: {address}!")
+            logging.exception(f"Eception while writing value={value} to register: {data_value._absolut_address}!")
             return False
         return True
 
-    def __read_holding_registers(self,address:int, check_connection:bool = True)->Tuple[bool,Any]:
+    def __read_holding_registers(self,address:int, count:int, check_connection:bool = True)->Tuple[bool,Any]:
         """Internal methode to read holding registers from modbus"""
         if check_connection and not self.is_connected:
             logging.error("Connection to modbus is not established!")
             return False, None
         try:
-            result = self._conn.read_holding_registers(address=address, unit=self._slave_id)
+            result = self._conn.read_holding_registers(address=address,count=count ,unit=self._slave_id)
```

Comment:
```suggestion
            result = self._conn.read_holding_registers(address=address,count=count ,slave=self._slave_id)
```

I got the following error with this code: 
`13:25:08 ERROR mixin:136 Please do not use unit=, convert to slave=.`
following the instructions I did not receive an error.

Reply:
I really don't like how the two different solarfocus systems behave differently, but yeah slave makes a lot more sense.